src/ome_zarr_models/_rfc5_transforms/coordinate_transformations.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from typing import Annotated, Literal
+
+from pydantic import Field, field_validator
+
+from ome_zarr_models._rfc5_transforms.axes import Axes
+from ome_zarr_models._utils import duplicates
+from ome_zarr_models.base import BaseAttrs
+
+
+class CoordinateSystem(BaseAttrs):
+    name: str
+    axes: Axes
+
+    @field_validator("name")
+    def name_must_not_be_empty(cls, value: str) -> str:
+        if value == "":
+            raise ValueError("name must be a non-empty string")
+        return value
+
+    @field_validator("axes")
+    def axes_must_not_be_empty(cls, value: Axes) -> Axes:
+        if len(value) == 0:
+            raise ValueError("axes must contain at least one axis")
+        return value
+
+    @field_validator("axes", mode="after")
+    @classmethod
+    def _ensure_unique_axis_names(cls, axes: Axes) -> Axes:
+        """
+        Ensures that the names of the axes are unique.
+        """
+        name_dupes = duplicates(a.name for a in axes)
+        if len(name_dupes) > 0:
+            msg = (
+                f"Axis names must be unique. Axis names {tuple(name_dupes.keys())} are "
+                "repeated."
+            )
+            raise ValueError(msg)
+        return axes
+
+
+class CoordinateTransformation(BaseAttrs):
```

Comment:
A few complications for `input` and `output`, based on a close read of the current version of the transforms PR:
- in the context of a `byDimension` transform, `input` / `output` could be a _list_ of strings. This may vastly complicate validation. See discussion [here](https://github.com/ome/ngff/pull/138/files#r2081982273).
- `input`/`output` are either a string / list of strings, or unset. So this type should include some logic that excludes those fields from JSON serialization if they are set to `None`.

Reply:
I agree, the `byDimension` case complicates the handling of `input` and `output`. I think we should deal with this in a separate PR. I moved this discussion in a issue: https://github.com/ome-zarr-models/ome-zarr-models-py/issues/191.
I will resolve this conversation.